controllers/glanceapi_controller.go

```diff
@@ -608,6 +608,12 @@ func (r *GlanceAPIReconciler) reconcileNormal(ctx context.Context, instance *gla
 
 	if instance.Status.ReadyCount > 0 {
```

Comment:
There is a slight variant of this check in couple of places[1]. 
If Instance.Spec.Replicas is set to 0  by the user then ReadyCount will be 0 and the GlanceAPI CR won't be set Ready even though the operator successfully applied the requested Specification to the Deployment resource. 

[1] e.g. https://github.com/openstack-k8s-operators/nova-operator/blob/532bc7f74c7f8893ad9ab53690bed442c843619b/controllers/novametadata_controller.go#L420-L432

Reply:
That's a good point.  The user's desired `Replicas` count should also be considered.  I will add that.

Reply:
Thanks @gibizer and @abays. It makes sense, and I agree we can easily cover this scenario via envTest.